pkg/apis/v1/nodepool_validation.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/robfig/cron/v3"
+	"github.com/samber/lo"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"knative.dev/pkg/apis"
+)
+
+func (in *NodePool) SupportedVerbs() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{
+		admissionregistrationv1.Create,
+		admissionregistrationv1.Update,
+	}
+}
+
+func (in *NodePool) Validate(_ context.Context) (errs *apis.FieldError) {
+	return errs.Also(
+		apis.ValidateObjectMetadata(in).ViaField("metadata"),
+		in.Spec.validate().ViaField("spec"),
+	)
+}
+
+// RuntimeValidate will be used to validate any part of the CRD that can not be validated at CRD creation
+func (in *NodePool) RuntimeValidate() (errs *apis.FieldError) {
```

Comment:
How much of this runtime validation is needed?

Reply:
I'm planning on adding validation changes in a separate PR